applications/audio/broadcast_config_tool/src/main.c

```diff
@@ -1258,6 +1279,252 @@ static int cmd_program_info(const struct shell *shell, size_t argc, char **argv)
+
+	if (argc != 4) {
+		shell_error(shell,
+			    "Usage: bct phy <1, 2 or 4 (coded)> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "PHY must be a digit");
+		return -EINVAL;
+	}
+
+	uint8_t phy = (uint8_t)atoi(argv[1]);
+
+	if (phy != BT_AUDIO_CODEC_QOS_1M && phy != BT_AUDIO_CODEC_QOS_2M &&
+	    phy != BT_AUDIO_CODEC_QOS_CODED) {
+		shell_error(shell, "Invalid PHY");
+		return -EINVAL;
+	}
+
+	ret = argv_to_indexes(shell, argc, argv, &big_index, 2, &sub_index, 3);
+	if (ret) {
+		return ret;
+	}
+
+	broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.phy = phy;
+	broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+
+	return 0;
+}
+
+static int cmd_framing(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_index;
+	uint8_t sub_index;
+
+	if (argc != 4) {
+		shell_error(shell,
+			    "Usage: bct framing <unframed/framed> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	ret = argv_to_indexes(shell, argc, argv, &big_index, 2, &sub_index, 3);
+	if (ret) {
+		return ret;
+	}
+
+	if (strcasecmp(argv[1], "unframed") == 0) {
+		broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.framing =
+			BT_AUDIO_CODEC_QOS_FRAMING_UNFRAMED;
+		broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+	} else if (strcasecmp(argv[1], "framed") == 0) {
+		broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.framing =
+			BT_AUDIO_CODEC_QOS_FRAMING_FRAMED;
+		broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+	} else {
+		shell_error(shell, "Invalid framing type");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int cmd_rtn(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_index;
+	uint8_t sub_index;
+
+	if (argc != 4) {
+		shell_error(shell, "Usage: bct rtn <num> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "RTN must be a digit");
+		return -EINVAL;
+	}
+
+	if (strtoul(argv[1], NULL, 10) > UINT8_MAX) {
+		shell_error(shell, "RTN must be less than %d", UINT8_MAX);
+		return -EINVAL;
+	}
+
+	uint8_t rtn = (uint8_t)atoi(argv[1]);
+
+	ret = argv_to_indexes(shell, argc, argv, &big_index, 2, &sub_index, 3);
+	if (ret) {
+		return ret;
+	}
+
+	broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.rtn = rtn;
+	broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+
+	return 0;
+}
+
+static int cmd_sdu(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_index;
+	uint8_t sub_index;
+
+	if (argc != 4) {
+		shell_error(shell, "Usage: bct sdu <octets> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "SDU must be a digit");
+		return -EINVAL;
+	}
+
+	if (strtoul(argv[1], NULL, 10) > UINT16_MAX) {
+		shell_error(shell, "SDU must be less than %d", UINT16_MAX);
```

Comment:
This is an unrealistically high value. Could we rather use a CONFIG_X which represents the actual value?

Reply:
This was discussed and we decided not to limit the end-user since this is a testing tool.